core/src/main/java/io/hyperfoil/core/util/BitSetResource.java

```diff
@@ -1,11 +1,88 @@
 package io.hyperfoil.core.util;
 
+import io.hyperfoil.api.session.Session;
+
 import java.util.BitSet;
 
-import io.hyperfoil.api.session.Session;
+public interface BitSetResource extends Session.Resource {
+
+    void set(int index);
+
+    boolean get(int index);
+
+    void clear(int index);
+
+    static BitSetResource with(int nbits) {
+        if (nbits == 0 || nbits == 1) {
+            class SingleBitSetResource implements BitSetResource {
```

Comment:
Can't we move `SingleBitSetResource` and `MultiBitSetResource` definition outside the `if-else` (either as static classes or even in a sub-package)? I am not a fan of class definition inside the flow but maybe there is a valid reason.

Reply:
The main reason was to avoid having 2 separate small classes just for few lines of codes.
Sadly java interfaces doesn't allow to define private classes within them, apart with this "trick" I have used (the anonymous/hidden definitions are still just 2)

Reply:
Said that I am not biased into doing like this at any cost, and instead I'm opened to move them into their separate files too

Reply:
No I think that's fine as it is, I was just curious and to be honest I missed that the outer class is an interface instead :smile: